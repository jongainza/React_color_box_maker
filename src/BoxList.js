import React, { useState } from "react";
import Box from "./Box";
import BoxForm from "./BoxForm";
import { v4 as uuidv4 } from "uuid";

const BoxList = () => {
  const INITIAL_VALUES = [
    {
      backgroundColor: "pink",
      width: "100",
      height: "100",
      id: uuidv4(),
      key: uuidv4(),
    },
    {
      backgroundColor: "olive",
      width: "35",
      height: "60",
      id: uuidv4(),
      key: uuidv4(),
    },
    {
      backgroundColor: "green",
      width: "100",
      height: "150",
      id: uuidv4(),
      key: uuidv4(),
    },
  ];
  const [boxes, setBoxes] = useState(INITIAL_VALUES);

  const addBox = (newBox) => {
    setBoxes((boxes) => [
      ...boxes,
      {
        backgroundColor: newBox.backgroundColor,
        height: newBox.heigth,
        width: newBox.width,
        id: uuidv4(),
      },
    ]);
  };
  return (
    <div>
      <h3>Boxes</h3>
      {boxes.map((b) => (
        <Box
          backgroundColor={b.backgroundColor}
          height={b.height}
          width={b.width}
          id={b.id}
          key={b.id}
        />
      ))}
      <BoxForm addBox={addBox} />
    </div>
  );
};

export default BoxList;
