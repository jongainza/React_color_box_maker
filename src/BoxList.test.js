import React from "react";
import { render, fireEvent, queryByLabelText } from "@testing-library/react";
import { getByStyle } from "@testing-library/user-event";

import BoxList from "./BoxList";

it("renders without crashing", function () {
  render(<BoxList />);
});

it("maches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a new box", function () {
  const { queryByText, getByLabelText, getByText, getByStyle } = render(
    <BoxList />
  );
  const inputLabel = queryByText("Background Color");
  expect(inputLabel).toBeInTheDocument();

  const input = getByLabelText("Background Color");
  const btn = queryByText("Add Box!");
  expect(getByStyle(document.body, { backgroundColor: "peach" })).toBeNull();

  fireEvent.change(input, { target: { value: "peach" } });
  fireEvent.click(btn);

  const newBox = getByStyle(document.body, { backgroundColor: "peach" });
  expect(newBox).toBeInTheDocument();
});
