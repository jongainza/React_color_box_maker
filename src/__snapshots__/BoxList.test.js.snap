// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Boxes
    </h3>
    <div
      style="background-color: pink; height: 100px; width: 100px;"
    />
    <div
      style="background-color: olive; height: 60px; width: 35px;"
    />
    <div
      style="background-color: green; height: 150px; width: 100px;"
    />
    <div>
      <form>
        <label
          for="backgroundColor"
        >
          Background Color
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          placeholder="Background Color"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          placeholder="Width"
          type="text"
          value=""
        />
        <label
          for="heigth"
        >
          Heigth
        </label>
        <input
          id="heigth"
          name="heigth"
          placeholder="Heigth"
          type="text"
          value=""
        />
        <p>
          Color: 
        </p>
        <p>
          Heigth: px
        </p>
        <p>
          Width: px
        </p>
        <button>
          Add Box!
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
